configs/example/gem5_library/checkpoints/riscv-hello-save-checkpoint.py

```diff
@@ -107,8 +107,8 @@
 
 # Lastly we run the simulation.
 max_ticks = 10**6
-simulator = Simulator(board=board, full_system=False)
-simulator.run(max_ticks=max_ticks)
+simulator = Simulator(board=board, full_system=False, max_ticks=max_ticks)
+simulator.run()
```

Comment:
Why this change? I think it makes a lot more sense to have the `max_ticks` in `run`, personally. See my comment below in the `run` function for more detail.